File: ITechQuiz/client/src/app/models/question.ts
import {Option} from "./option";

export class Question{
    public constructor(
        public id?:string,
        public title?:string,
        public multiple?:boolean,
        public maxSelected?:number,
        public required?:boolean,
        public surveyId?:string,
        public options?: Option[]
    )
    { }
}

File: ITechQuiz/Models/Question.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ITechQuiz.Models
{
    public class Question
    {
        [Required]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Title { get; set; }

        [Required]
        public bool Multiple { get; set; }

        [Required]
        public int MaxSelected { get; set; }

        [Required]
        public bool Required { get; set; }

        public ICollection<Option> Options { get; set; }

        public Survey Survey { get; set; }

        public Guid SurveyId { get; set; }

    }
}
